draft-ietf-quic-transport.md

```diff
@@ -3940,6 +4007,19 @@ described in {{QUIC-TLS}}.  This protection does not provide confidentiality or
 integrity against on-path attackers, but provides some level of protection
 against off-path attackers.
 
+Additionally, the token is accompanied by a checksum.  This is because when a
```

Comment:
Seeing a checksum here makes me uncomfortable. We're adding a checksum to retry packets because they have no other mechanism to detect corruption. Initial packets on the other hand have an auth tag, so a corrupted token will cause the deprotection to fail. In an ideal world we should handle that like any other corruption on the initial packet. But I think this goes back to my other question about version negotiation, I think there be dragons.

Reply:
I second the question here; the checksum doesn't appear to serve a purpose not already covered here.

Reply:
We need Token Checksum so that the server can easily determine if there was a bit flip on the wire when it receives an invalid token.

The PR (up until of 7eded0a) requires the server to drop the packet if there was a bit flip on the wire. OTOH, if it was an unusable token (rather than a bit flip), the server responds with a VN. This distinction is necessary to prevent bit flips from being mis-detected as an downgrade attack (see https://github.com/quicwg/base-drafts/pull/3166#discussion_r340344021).

That said, I agree that this change is intrusive. Considering that we need to require certain properties for the token (see https://github.com/quicwg/base-drafts/pull/3166#discussion_r340396154), we might as well state that a NEW_TOKEN token should be generated in a way that the server can detect bit flips even it has lost the keys to decrypt the token. In practice, that can be easily achieved by attaching CRC32 (or zero-key GMAC).

But finally, this entire discussion is based on the assumption that we need a downgrade path, which @martinthomson disputes in https://github.com/quicwg/base-drafts/pull/3166#pullrequestreview-308948794.

Reply:
The checksum is very useful in situations where a bridge forwards a connection over an alternative medium such as USB or Bluetooth. The Retry and VN packets are the only packets that don't have self-integrity checks and such a bridge would be forced to provide additional framing. This framing is tolerable for VN because it is universal, but for Retry it requires version knowledge. It is much simpler if packets can just be forwarded with no knowledge other than length.

Reply:
The checksum can be provided by the server on its own.  The only reason you would provide a separate field for one is so that the client can validate it.  But the client can ensure that the NEW_TOKEN was not modified through packet protection.  Good to see this was removed :)